packages/frontend/src/hooks/useChartColorConfig.ts

```diff
@@ -0,0 +1,97 @@
+import { FeatureFlags } from '@lightdash/common';
+import { useFeatureFlagEnabled } from 'posthog-js/react';
+import { useCallback } from 'react';
+
+/**
+ * Generic series-like object that contains the relevant portions used
+ * for color assignment. Requires the caller to unpack whatever series
+ * format it has into this compatible one.
+ */
+interface SeriesLike {
```

Comment:
Opted to use a series-like object as the argument, since there's inconsistencies between the reported/actual series objects passed around for different chart types. This forces the caller to normalize the data.